pkg/scheduler/framework/runtime/framework.go

```diff
@@ -1421,29 +1421,70 @@ func (f *frameworkImpl) RunPreBindPlugins(ctx context.Context, state fwk.CycleSt
 		logger = klog.LoggerWithName(logger, "PreBind")
 		logger = klog.LoggerWithValues(logger, "node", klog.ObjectRef{Name: nodeName})
 	}
+
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	statusCh := parallelize.NewErrorChannel[*fwk.Status]()
+
+	pluginGroups := f.getPreBindPluginGroups(state)
+	for _, plugins := range pluginGroups {
+		f.Parallelizer().Until(ctx, len(plugins), func(index int) {
+			pl := plugins[index]
+			ctx := ctx
+			if verboseLogs {
+				logger := klog.LoggerWithName(logger, pl.Name())
+				ctx = klog.NewContext(ctx, logger)
+			}
+			plStatus := f.runPreBindPlugin(ctx, pl, state, pod, nodeName)
+			if !plStatus.IsSuccess() {
+				if plStatus.IsRejected() {
+					logger.V(4).Info("Pod rejected by PreBind plugin", "pod", klog.KObj(pod), "node", nodeName, "plugin", pl.Name(), "status", plStatus.Message())
+					plStatus.SetPlugin(pl.Name())
+					statusCh.SendErrorWithCancel(plStatus, cancel)
+					return
+				}
+				err := plStatus.AsError()
+				logger.Error(err, "Plugin failed", "plugin", pl.Name(), "pod", klog.KObj(pod), "node", nodeName)
+				statusCh.SendErrorWithCancel(fwk.AsStatus(fmt.Errorf("running PreBind plugin %q: %w", pl.Name(), err)), cancel)
+			}
+		}, metrics.PreBind)
+		if status := statusCh.ReceiveError(); status != nil {
+			return status
+		}
+	}
+	return statusCh.ReceiveError()
+}
+
+// getPreBindPluginGroups chunks the plugins implementing the PreBind extension point
+// into groups of plugins that can be run in parallel.
+// The returned value consists of non-skipped plugins only.
+func (f *frameworkImpl) getPreBindPluginGroups(state fwk.CycleState) [][]fwk.PreBindPlugin {
+	var groups [][]fwk.PreBindPlugin
+	group := []fwk.PreBindPlugin{}
```

Comment:
Since you use append to add the plugins, you don't need to initialize:
```suggestion
	var group []fwk.PreBindPlugin
```